File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        Set = set(nums)
        
        maxLength = 0
        
        for i in nums:
            if (i - 1) not in Set:
                longestSequence = [i]
            
                while (i + 1) in Set:
                    longestSequence.append(i + 1)
                    i += 1

                if len(longestSequence) > maxLength:
                    maxLength = len(longestSequence) 
            


        return maxLength

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashSet<Integer> duplicates = new HashSet<Integer>();
        for (int i = 0; i < nums.length; i++){
            if (duplicates.contains(nums[i])){
                return true;
            } else {
                duplicates.add(nums[i]);
            }
        }
        return false;
    }
}
